src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp

```diff
@@ -282,4 +286,66 @@ void PCRPrimerDesignForDNAAssemblyOPWidget::sl_loadOtherSequenceInPcr() {
 
 }
 
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_onFindTaskFinished() {
+    CHECK(sender() == pcrTask, );
+    SAFE_POINT(nullptr != pcrTask, L10N::nullPointerError("InSilicoPcrTask"), );
+    if (pcrTask->isCanceled() || pcrTask->hasError()) {
+        disconnect(pcrTask, SIGNAL(si_stateChanged()));
+        pcrTask = nullptr;
+        setEnabled(true);
+        return;
+    }
+    CHECK(pcrTask->isFinished(), );
+    //do nothing in case of empty result
+    foreach (const U2Region &region, pcrTask->getResults()) {
+        if (region != U2Region()) {
+            showResults();
```

Comment:
There is no need to call this function an each result - just call it one time if results aren't empty

Reply:
It called only once, there is `return;` after it.

Reply:
Yeah, now see it